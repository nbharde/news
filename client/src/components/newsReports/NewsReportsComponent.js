import React from "react";
import { useSelector } from "react-redux";

import NewsReportComponent from "../newsReport/NewsReportComponent";
import SearchComponent from "../search/SearchComponent";

const NewsReportsComponent = (props) => {
    const { articles } = useSelector(state => state.newsReportsReducer);

    return(
        <div className="container">
            <SearchComponent />

            { articles && articles.length > 0 ? (
                <div className="content-container" data-testid="articles_testId">
                    {articles.map(article => (
                        <NewsReportComponent article={article} key={article.title}/>
                    ))}
                </div>
            ) : 
            <div data-testid="no_article_testId">No article found</div>
            }
        </div>
    );
}

export default NewsReportsComponent