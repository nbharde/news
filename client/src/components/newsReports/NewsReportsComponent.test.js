import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';

import NewsReportsComponent from "./NewsReportsComponent";
import { useSelector, useDispatch } from "react-redux";

jest.mock("react-redux", () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

const mockAppState = { newsReportsReducer: {
    articles: [{
        title: "COVID-19: UK reports",
        description: "Description",
    }]
}}

describe("NewsReportsComponent", () => {

    beforeEach(() => {
        useSelector.mockImplementation(callback => {
        return callback(mockAppState);
        });

        const dispatch = jest.fn();
        useDispatch.mockReturnValue(jest.fn());
    });

    afterEach(() => {
        useSelector.mockClear();
        cleanup;
    });

    it("should render a news reports", () => {
        const { queryByTestId } = render(<NewsReportsComponent />);
        expect(queryByTestId("articles_testId")).not.toBeNull();
    });

    it("should not render if articles is empty", () => {
        const localMockState = {"newsReportsReducer": {}};
        useSelector.mockImplementation(callback => {
            return callback(localMockState);
        });

        const { queryByTestId } = render(<NewsReportsComponent />);
        expect(queryByTestId("no_article_testId")).not.toBeNull();

    });

    it("matches snapshot", () => {
        const { container } = render(<NewsReportsComponent />);
        expect(container).toMatchSnapshot();
    });
});
