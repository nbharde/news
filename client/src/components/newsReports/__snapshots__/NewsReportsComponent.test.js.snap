// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsReportsComponent matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="searchbox-container"
      data-testid="searchbox_testId"
    >
      <form
        autocomplete="off"
        data-testid="form_testId"
      >
        <input
          class="search-box"
          data-testid="input_testId"
          placeholder="Enter Keyword to Search"
          type="text"
        />
        <button
          alt="Search"
          class="submit-btn"
          data-testid="submit_btn_testId"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      class="content-container"
      data-testid="articles_testId"
    >
      <div
        class="article"
        data-testid="headline_testId"
      >
        <div
          class="image"
        >
          <img
            alt="News"
            height="100"
            width="130"
          />
        </div>
        <div
          class="content"
        >
          <div
            class="title"
          >
            COVID-19: UK reports
          </div>
          <div
            class="description"
          >
            Description
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
